localstack/services/apigateway/invocations.py

```diff
@@ -477,13 +478,6 @@ def invoke_rest_api_integration_backend(invocation_context: ApiInvocationContext
                 table.put_item(Item=event_data)
                 response = requests_response(event_data)
                 return response
-        if uri.startswith("arn:aws:apigateway:") and ".appsync-api:" in uri:
```

Comment:
Consider adding a comment that `invoke_rest_api_integration_backend` is patched in ext (before moving to better abstractions; e.g., strategy pattern with a plugin-based factory).
In the long term, we should think about how to handle conditional pro extensions as this keeps reoccurring.

Reply:
👍 for the planned refactoring `# TODO: refactor this to have a class per integration type to make it easy to test the encapsulated logic`
This looks hard to test and maintain at first glance :)

Reply:
This is actually how it is being implemented now. Each backend integration has a class with a set of steps that are always required, request parameters resolvers, request templates, response parameters resolvers, and response templates.

Happy to hear your thoughts!